src/lib.rs

```diff
@@ -0,0 +1,54 @@
+use serde_json_core;
+
+pub trait StringSet {
+    fn string_set(&mut self, topic_parts:
+            core::iter::Peekable<core::str::Split<char>>, value: &str) ->
+            Result<(),()>;
+
+}
+
+macro_rules! derive_primitive {
```

Comment:
I don't think we need to derive these... Primitives should never call `string_set()`, but rather justy call `serde_json_core::from_str()` directly, shouldn't they? Even if someone attempts to call set("a/b", 3), where "a" is a primative, an error will be generated.

Reply:
After further review, I see I'm wrong here due to the recursion mechanism.